### DISCUSSION TITLE
Can UWB tag be used to transmit long data packets to anchor?

### DISCUSSION THREAD

**Discussion by lalalalang999** [bitcraze/discussions#644] [Q&A]
I have an idea that UWB can do positioning, and I also hope to transmit data while positioning, and the tag transmits large packets of data (1024KB) to the base station. I saw in carzfile-firmware that [sendLppShort ](https://github.com/bitcraze/crazyflie-firmware/blob/6490515f168733c89ee97224ec65b546676503de/src/deck/drivers/src/lpsTdoa2Tag.c#:~:text=static%20%20void%20%20sendLppShort%20(dwDevice_t%20*dev%2C%20lpsLppShortPacket_t%20*packet))is used to transmit short data packets, and in the DW1000 user manual, I saw that a data packet of up to 1024 bytes can be transmitted at one time. May I ask if I transmit a large-capacity data packet (transmitting 1024KB every 100ms) from the tag to the anchor, will it affect the positioning accuracy?

**Comment by krichardsson** [comment]
I'm not exactly sure about the details of longer packets, configuration and so on, but just from a positioning point of view I do not think it would be a problem. The best way is probably to try it out :-)

One way to look at it is that sending a long packet probably takes a while, this would be similar to loosing some packet which is not a big problem. As long as your useful packet rate is high enough it should be fine. I'm not exactly sure what rate is required (depends on mode, system geometry, requirements and so on) but 100 - 200 packets/s is maybe OK?

**Reply by lalalalang999** [reply]
Thanks for your reply, I'll try it out and get back to you with results

**Reply by lalalalang999** [reply]
hi:
I tried to send lpp short packets from tag to anchor under TDOA2 algorithm, and I found that this affects the accuracy of tag's localization. I send packets to anchor via [sendLppShort](https://github.com/bitcraze/crazyflie-firmware/blob/6490515f168733c89ee97224ec65b546676503de/src/deck/drivers/src/lpsTdoa2Tag.c#LL155C13-L155C25) tag code, and [handleServicePacket](https://github.com/bitcraze/lps-node-firmware/blob/73f8960c728a3030b331b0abe19f73239bf6f38b/src/uwb_tdoa_anchor2.c#L382) in anchor to receive and process packets, I see that re-synchronization of ANCHOR is done every time packets are received in [ctx.state = syncTdmaState;](https://github.com/bitcraze/lps-node-firmware/blob/73f8960c728a3030b331b0abe19f73239bf6f38b/src/uwb_tdoa_anchor2.c#L384) ,and the description of the comment is "The service packet handling time desynchronized us, lets resynch". This will affect the positioning accuracy by resynchronizing every time a packet is received, because I envision a communication where each ANCHOR will receive a packet from tag every 2.1ms. My question is whether the ANCHOR can receive a short lpp packet from tag without resynchronizing the state, or whether there is a better solution idea.

**Reply by ataffanel** [reply]
My assumption of what is happening is that you are sending long enough packet to overflow the timeslot window. On easy way to test this theory would be to increase the timeslot window size. This can be done by changing the [`TDMA_SLOT_BITS`](https://github.com/bitcraze/lps-node-firmware/blob/73f8960c728a3030b331b0abe19f73239bf6f38b/src/uwb_tdoa_anchor2.c#LL67C9-L67C23) constant to 27. This needs to be done both on the Crazyflie and on the anchor and it will set a timeslot of ~4ms instead of 2ms, effectively dividing by 2 the update rate.

Timeslot length are currently lock at power-of-two values by design. It would be possible to change that but would require more change in the anchors and in the Crazyflie.

**Reply by lalalalang999** [reply]
Is this window time slot now 2.1ms. If I can ensure that the data transfer time does not overflow the window time slot, then I don't need to synchronize every time I receive a packet. What I want is to reduce the amount of data transferred by compressing it from the previous 1024 bytes to 120 bytes.
